File: server/feed2/src/main/scala/info/glennengstrand/news/Link.scala
package info.glennengstrand.news

object Link {
  val path = raw"/participant/([0-9]+)".r
  def extractId(link: String): Long = {
    link match {
      case path(id) => id.toLong
      case _ => link.toLong
    }
  }
  def toLink(id: Long): String = {
    "/participant/" + id.toString
  }
}

File: server/feed8/src/main/java/info/glennengstrand/util/Link.java
package info.glennengstrand.util;

import java.util.regex.Pattern;
import java.util.regex.Matcher;

public abstract class Link {
	private static final Pattern participant = Pattern.compile("/participant/([0-9]+)");
	public static Long extractId(String value) {
		Matcher m = participant.matcher(value);
		if (m.matches()) {
			return Long.parseLong(m.group(1));
		} else {
			return Long.parseLong(value);
		}
	}
	public static String toLink(Long id) {
		return "/participant/".concat(String.valueOf(id));
	}
}
